internal/manager/run.go

```diff
@@ -210,6 +217,26 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic, d
 		}
 	}
 
+	// TODO https://github.com/Kong/kubernetes-ingress-controller/issues/3922
+	// This requires the Konnect client, which currently requires c.Konnect.ConfigSynchronizationEnabled also.
+	// We need to figure out exactly how that config surface works. Initial direction says add a separate toggle, but
+	// we probably want to avoid that long term. If we do have separate toggles, we need an AND condition that sets up
+	// the client and makes it available to all Konnect-related subsystems.
+	if c.Konnect.LicenseSynchronizationEnabled {
+		konnectAPIClient, err := konnect.NewLicenseAPIClient(c.Konnect)
```

Comment:
similar to above comment, could we rename this to `konnectLicenseClient` or `konnectLicenseAPIClient`?

Reply:
Eh, whatever, these are all going to get refactored for https://github.com/Kong/kubernetes-ingress-controller/issues/3921.